api/src/main/java/jakarta/data/repository/Find.java

```diff
@@ -55,23 +73,45 @@
  * instances with the given value of the {@code model} field.
  * </p>
  *
- * <p>A method annotated with {@code @Find} must return one of the following types:</p>
+ * <p>A method annotated with {@code @Find} must return one of the following types,
+ * where {@code E} is the entity type, entity attribute type, or record type inferred or explicitly specified by the {@code @Find} annotation:</p>
  * <ul>
- *     <li>an entity type {@code E}, when the method returns a single instance,</li>
+ *     <li>{@code E}, when the method returns a single instance</li>
  *     <li>{@code Optional<E>}, when the method returns at most a single instance,</li>
- *     <li>an entity array type {@code E[]},
+ *     <li>an array type {@code E[]},
  *     <li>{@code List<E>},</li>
- *     <li>{@code Stream<E>}, or</li>
- *     <li>{@code Page<E>} or {@code CursoredPage<E>}.</li>
+ *     <li>{@code Stream<E>},</li>
+ *     <li>{@code Page<E>}, or</li>
+ *     <li>{@code CursoredPage<E>} (only allowed when {@code E} is the entity type).</li>
  * </ul>
  *
- * <p>An automatic query method annotated {@code Find} returns an entity instance for every record which satisfies the
- * parameter-based conditions.</p>
+ * <p>For example, if a {@code Car} entity has attribute names including {@code make},
```

Comment:
What happens if I have the same field?

```java

@Entity 
public class Person {

private CreditCard card;
private String name;
}

@Embedded
public class CreditCard {
private String name;
private String number;
}
```

```java
record ModelInfo(String name) {}
```

In this case, we have both with `name.` Which one should I take?

Reply:
We already have a convention for this which is to use `card_name` for fields of the embeddable.